server/server.ts

```diff
@@ -107,6 +126,93 @@ app.get('/ping', (_req, res) => {
   res.json('pong');
 });
 
+app.post('/discord/auth', async (req, res) => {
+  const decoded = await validateUserToken(req.body?.uid, req.body?.token);
+  if (!decoded) {
+    return res.status(400).json({ error: 'invalid user token' });
+  }
+  const customer = await getCustomerByEmail(decoded.email as string);
+  const isSubscriber = Boolean(
+    customer?.subscriptions?.data?.find((sub) => sub?.status === 'active')
+  );
+  if (!isSubscriber) {
+    return res.status(400).json({ error: 'not subscribed' });
+  }
+  if (!postgres || !discordBot) {
+    return res.sendStatus(500);
+  }
+  try {
+    const user = await discordBot.assignRole(
+      req.body?.username,
+      req.body?.discriminator
+    );
+    if (!user) {
+      return res.status(400).json({
+        error: 'Discord account not found. Please join our Discord server.',
+      });
+    }
+    // Allow only one Discord account with role per email.
+    // Check if this customer already has a discord account with a role. If so remove that role.
+    const result = await postgres?.query(
+      'SELECT * FROM discord WHERE email = $1',
+      [decoded.email]
+    );
+    if (result.rowCount) {
+      await discordBot.assignRole(
```

Comment:
This is the link action? we don't need to remove any roles?

Reply:
if you pass the argument undo=true to assignRole it removes the role. Maybe this is a bit confusing?

Reply:
Yeah i'm asking why we need to remove the role here

Reply:
Oh I see. Ok we will leave out all role removing stuff.

Reply:
To answer the question: We remove the role here because we want to limit the linked discord accounts per email to max 1 . if we dont remove the role here someone could theoretically assign a sub role to an unlimited number of accounts.

Would that be fine?

Reply:
But we defer this cleanup to later, right?